server/src/main/java/org/apache/druid/server/QueryLifecycle.java

```diff
@@ -102,6 +103,8 @@
 
   @MonotonicNonNull
   private Query<?> baseQuery;
+  @MonotonicNonNull
+  private Map<String, Object> userContext;
```

Comment:
can we store the keys themselves instead of storing the map?

Reply:
Great suggestion. Done.